src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -261,6 +272,11 @@ public static Transaction fromV1EnvelopeXdr(
       mOperations[i] = Operation.fromXdr(accountConverter, envelope.getTx().getOperations()[i]);
     }
 
+    Optional<SorobanTransactionData> sorobanData = Optional.absent();
+    if (envelope.getTx().getExt().getDiscriminant() == 1) {
+      sorobanData = Optional.of(envelope.getTx().getExt().getSorobanData());
```

Comment:
if choose to not use Optional as [mentioned earlier](https://github.com/stellar/java-stellar-sdk/pull/481/files#r1270983434), these later references can be simplified, as only need to set `sorobanData` if ext has it.

Reply:
Fixed in 37e0e19accdadde748a715af4e5dbf25a73ad40e